python/private/pypi/parse_simpleapi_html.bzl

```diff
@@ -96,7 +96,25 @@ def parse_simpleapi_html(*, url, content):
         whls = whls,
     )
 
+def _get_root_directory(url):
+    scheme_end = url.find("://")
+    if scheme_end == -1:
+        fail("Invalid URL format")
+
+    scheme = url[:scheme_end]
+    host_end = url.find("/", scheme_end + 3)
+    if host_end == -1:
+        host_end = len(url)
+    host = url[scheme_end + 3:host_end]
+
+    return "{}://{}".format(scheme, host)
+
 def _absolute_url(index_url, candidate):
```

Comment:
I think it may be better to pass the `index_url_root` to the `_absolute_url` instead of getting the root directory all the time within this function. Since this may be called many times, I am worried that we will be doing many string manipulations for nothing.

It would be also good to have a test so that this does not regress, but I  am not sure if it is easy to do with the current structuring of the code.

Reply:
Yes, the current implementation isn't the fastest. But do we really care about this here? While these string manipulations do require some CPU time, it's not significant given how fast modern CPUs are. Adding another argument to this function also makes the code feel less elegant.

For the test, I will add if possible. 

Thanks for the review.